docs/rules/amd-function-arity.md

```diff
@@ -0,0 +1,242 @@
+# Ensure AMD callbacks have correct number of parameters (amd-function-arity)
+
+
+## Rule Details
+
+This rule aims to ensure that AMD callbacks have the correct number of parameters in order to avoid confusion and maintainability issues.
+
+### Options
+
+This rule takes one option, an object with the key `allowExtraDependencies` and a Boolean value (default is `false`).
+
+```json
+    "requirejs/enforce-define": [2, { "allowExtraDependencies": true }]
```

Comment:
Open to suggestions. To me it's immaterial- the main purpose of the extra
dependencies is for side effects, but there could be other reasons. And
yes, I'll fix the wrong rule name, oops.
On Dec 11, 2015 9:36 PM, "Casey Visco" <notifications@github.com> wrote:

> In docs/rules/amd-function-arity.md
> <https://github.com/cvisco/eslint-plugin-requirejs/pull/46#discussion_r47426362>
> :
>
> > @@ -0,0 +1,242 @@
> > +# Ensure AMD callbacks have correct number of parameters (amd-function-arity)
> > +
> > +
> > +## Rule Details
> > +
> > +This rule aims to ensure that AMD callbacks have the correct number of parameters in order to avoid confusion and maintainability issues.
> > +
> > +### Options
> > +
> > +This rule takes one option, an object with the key `allowExtraDependencies` and a Boolean value (default is `false`).
> > +
> > +```json
> > +    "requirejs/enforce-define": [2, { "allowExtraDependencies": true }]
>
> Also I'm not 100% sold on the naming of this option. It's more about
> allowing require of dependencies for side effects, no?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cvisco/eslint-plugin-requirejs/pull/46/files#r47426362>
> .
>